exasol_advanced_analytics_framework/testing/mock_query_handler_runner.py

```diff
@@ -0,0 +1,102 @@
+                     QueryHandler[ParameterType, ResultType]]):
+        self._sql_executor = sql_executor
+        query_handler = query_handler_factory(parameter, top_level_query_handler_context)
+        self._state = QueryHandlerRunnerState(
+            top_level_query_handler_context=top_level_query_handler_context,
+            query_handler=query_handler
+        )
+
+    def run(self) -> ResultType:
+        try:
+            result = self._state.query_handler.start()
+            if isinstance(result, Finish):
+                self.handle_finish()
+                return result.result
+            else:
+                while isinstance(result, Continue):
+                    result = self.handle_continue(result)
+                if isinstance(result, Finish):
+                    self.handle_finish()
+                    return result.result
+                else:
+                    raise RuntimeError("Unknown Result")
+        except Exception as e:
+            self.handle_finish()
+            raise RuntimeError(f"Execution of query handler {self._state.query_handler} failed.") from e
+
+    def handle_continue(self, result: Continue):
+        self.release_and_create_new_return_query_query_handler_context()
+        self.cleanup_query_handler_context()
+        for query in result.query_list:
+            self._sql_executor.execute(query.query_string)
+        input_query_result = self.run_input_query(result)
+        result = self._state.query_handler.handle_query_result(input_query_result)
+        return result
+
+    def run_input_query(self, result: Continue):
+        input_query_view, input_query = self._wrap_return_query(result.input_query)
+        self._sql_executor.execute(input_query_view)
+        input_query_result_set = self._sql_executor.execute(input_query)
+        if input_query_result_set.columns() != result.input_query.output_columns:
+            raise RuntimeError(f"Specified columns {result.input_query.output_columns} of the input query "
+                               f"are equal to the actual received columns {input_query_result_set.columns()}")
+        input_query_result_table = input_query_result_set.fetchall()
+        input_query_result = MockQueryResult(data=input_query_result_table,
+                                             columns=result.input_query.output_columns)
+        return input_query_result
+
+    def handle_finish(self):
+        if self._state.input_query_query_handler_context is not None:
+            self._state.input_query_query_handler_context.release()
+        self._state.top_level_query_handler_context.release()
+        self.cleanup_query_handler_context()
+
+    def cleanup_query_handler_context(self):
+        cleanup_query_list = \
+            self._state.top_level_query_handler_context.cleanup_released_object_proxies()
+        for query in cleanup_query_list:
+            self._sql_executor.execute(query.query_string)
```

Comment:
This code snippet appears several times in this class, create helper function:
def execute_query_list(query_list: Iterator[str]) -> None:
       for query in query_list:
            self._sql_executor.execute(query.query_string)

Reply:
I would prefer not to do this, because query list has a dedicated meaning in this algorithm. We can rename the method to execute_queries or so, then it should be possible

Reply:
done